cli/cli-client/pom.xml

```diff
@@ -118,12 +127,135 @@
+                        <phase>package</phase>
+                        <goals>
+                           <goal>build-classpath</goal>
+                        </goals>
+                        <configuration>
+                           <outputProperty>maven.compile.classpath</outputProperty>
+                           <pathSeparator>:</pathSeparator>
+                           <includeScope>compile</includeScope>
+                        </configuration>
+                     </execution>
+                  </executions>
+                  <configuration>
+                     <outputDirectory>${project.build.directory}/lib</outputDirectory>
+                     <silent>true</silent>
+                  </configuration>
+               </plugin>
+               <plugin>
+                  <groupId>org.codehaus.mojo</groupId>
+                  <artifactId>exec-maven-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <id>generate-native-reflection</id>
+                        <phase>package</phase>
+                        <goals>
+                           <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                           <executable>java</executable>
+                           <workingDirectory>target</workingDirectory>
+                           <arguments>
+                              <argument>-cp</argument>
+                              <classpath/>
+                              <argument>org.aesh.util.GraalReflectionGenerator</argument>
+                              <argument>${cli.main.class}</argument>
+                           </arguments>
+                        </configuration>
+                     </execution>
+                     <execution>
+                        <id>native</id>
+                        <phase>package</phase>
+                        <goals>
+                           <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                           <executable>${env.GRAALVM_HOME}/bin/native-image</executable>
+                           <workingDirectory>${project.build.directory}</workingDirectory>
+                           <arguments>
+                              <argument>--verbose</argument>
+                              <argument>-H:+ReportUnsupportedElementsAtRuntime</argument>
+                              <argument>-H:+ReportExceptionStackTraces</argument>
+                              <argument>-H:ReflectionConfigurationFiles=cli_reflection.json</argument>
+                              <argument>-cp</argument>
+                              <argument>${maven.compile.classpath}</argument>
+                              <argument>-jar</argument>
+                              <argument>${project.build.finalName}.jar</argument>
+                              <argument>infinispan</argument>
```

Comment:
This is actually a little more complex: I don't want to make commands "branded" as it makes things unportable between the two sides. In WildFly/EAP the CLI is called `jboss-cli.sh` and does not change. Using the upstream name here (or a neutral one) is probably the best. Bear in mind that we will also have to create an alias/symbolic link to a `kubectl-infinispan` (or whatever) so that it can be invoked as a slave CLI to `kubectl`/`oc`